### ISSUE TITLE
Installing from source causes `cargo install` PATH warning on Windows due to path canonicalization

### ISSUE THREAD

**Issue by CAD97** [nabijaczleweli/cargo-update#252]
Example:

```
> cargo install-update -i cargo-docs-rs
    Polling registry 'https://index.crates.io/'.

Package        Installed  Latest   Needs update
cargo-docs-rs  No         v0.1.10  Yes

Installing cargo-docs-rs
 INFO resolve: Resolving package: 'cargo-docs-rs'
 WARN The package cargo-docs-rs v0.1.10 will be installed from source (with cargo)
    Updating crates.io index
  Installing cargo-docs-rs v0.1.10
    Updating crates.io index
    Finished release [optimized] target(s) in 0.59s
  Installing \\?\D:\.rust\cargo\bin\cargo-docs-rs.exe
   Installed package `cargo-docs-rs v0.1.10` (executable `cargo-docs-rs.exe`)
warning: be sure to add `\\?\D:\.rust\cargo\bin` to your PATH to be able to run the installed binaries
 INFO Cargo finished successfully
 INFO Done in 2.2362273s


Updated 1 package.
Overall updated 1 package: cargo-docs-rs.
```

The UNC-style path `\\?\D:\.rust\cargo\bin` is not on my PATH, but `D:\.rust\cargo\bin` is. Cargo (and `cargo binstall`) correctly is not displaying this warning itself; this warning is coming from the `cargo install-update` wrapper.

**Comment by nabijaczleweli** [comment]
Looks like `cargo-binstall` output. Can you repro this with `cargo-binstall --roots {whatever your cargo dir is configured to} --no-confirm --version =0.1.10 --force --quiet cargo-docs-rs`?

**Comment by NobodyXu** [comment]
> warning: be sure to add `\\?\D:\.rust\cargo\bin` to your PATH to be able to run the installed binaries

I think this is from cargo-install.

Cargo-binstall output looks like this


> INFO Cargo finished successfully

It's always `<log_level> <msg>`

**Comment by CAD97** [comment]
QuickInstall is now providing cargo-docs-rs@0.1.10 so I can't easily reproduce with this crate again. However, I have previously verified that neither `cargo install` nor `cargo binstall` print the warning on their own, only when orchestrated through `cargo install-update`.

**Comment by CAD97** [comment]
Oh and for clarity, my `$CARGO_HOME` is `D:\.rust\cargo`, i.e. not in the canonicalized extended-length format. This is probably due to a `path.canonicalize()` somewhere in cargo-install-update, where it really should only be doing `CARGO_HOME.join("bin")` without any canonicalization.

**Comment by nabijaczleweli** [comment]
Yeah, that was gonna be my guess as well – we run `cargo-binstall --roots {cargo-dir} ...` or `cargo install --root {cargo-dir} ...`, and `cargo-dir` is canonicalised, which yields the `\\?\` prefix.

I was gonna say I hadn't seen this warning before, even with
```
>echo %PATH% | tr ';' '\n' | grep cargo
D:\Users\nabijaczleweli\.cargo\bin
d:\Users\nabijaczleweli\.cargo\bin
```
but I just reproed it with `cargo install --root \\?\D:\Users\nabijaczleweli\.cargo treesize` so maybe I just never noticed it or whatever.

**Comment by nabijaczleweli** [comment]
@CAD97 can you try the current `master` branch (at least 3cc63719ab83cf9760fd4e09cf72f9eb9136771a)?

**Comment by nabijaczleweli** [comment]
Fix released in [`v13.4.0`](https://github.com/nabijaczleweli/cargo-update/releases/tag/v13.4.0)
